cpp/llm_chat.cc

```diff
@@ -615,7 +615,7 @@ class LLMChat {
     std::vector<int32_t> encoded = this->tokenizer_->Encode(all_prompt);
     tokens.insert(tokens.end(), encoded.begin(), encoded.end());
     if (this->sliding_window_ != -1 ||  // There is no max window size if we use sliding window
-        this->total_seq_len_ + tokens.size() + gen_mean_gen_len < this->max_window_size_) {
+        this->total_seq_len_ + (int)tokens.size() + gen_mean_gen_len < this->max_window_size_) {
```

Comment:
use `static_cast<int64_t>(tokens.size())`

Reply:
ok. I think this is a quite serious bug that has troubled me for more than two weeks. The quantized int4 version of rwkv5 seems to give very unintelligent responses, and it was only today that I thought to print out the prompt. Then I discovered that all the code after line 618 was ineffective, and finally pinpointed this issue. Now the quantized int4 version of rwkv5 can also generate text normally. The performance has also improved in other modes for rwkv.

Reply:
can you elaboratae a bit why static cast int is needed here? do we involve some negative numbers in computing this ?

Reply:
<img width="516" alt="图片" src="https://github.com/mlc-ai/mlc-llm/assets/35585791/a7da3687-7c90-4eab-91c0-afb5eec3e1fa">

Reply:
This error might occur whenever the input prompt is relatively long. Before using `static_cast<int64_t>` to convert `tokens.size()`, the expression `(this->total_seq_len_ + tokens.size() + gen_mean_gen_len)` might have experienced integer overflow at some stage, causing the result to be incorrectly interpreted as a negative number, which in turn erroneously returned `true` for the comparison operation.

Reply:
OK, I feel this is a strange way to think about it, given `max_window_size_ == -1`, we should specially check it, and that means there is no out of bound and we do not need to re-encode (aka running the code after), would be good for @Hzfengsy to take a loo as well

Reply:
Okay, there seems to be a bug in the handling of the rwkv system prompts. I expect that each interaction with the rwkv model should include the system prompt along with the current text. This is because its series of models(rwkv4,5,6) have higher requirements for prompts. Currently, only the first round of dialogue includes the system's prompt, and the system prompt is forgotten in subsequent dialogues